src/leader.py

```diff
@@ -0,0 +1,75 @@
+import time
+import urllib
+from concurrent.futures import ThreadPoolExecutor
+
+class LeaderManager:
```

Comment:
It will be great if you could provide a comment for this class. The name seems unclear to me.

Reply:
Would just BullyAlgorithm be more clear? Or perhaps LeaderElector?

Reply:
Maybe `BullyAlgorithm` would be better.